### PR TITLE
chore: Add v1 APIs and Conversion webhooks for v0.35.x

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Add v1 APIs and the conversion webhooks
- This is a copy and from main to release a patch version of 0.35

**How was this change tested?**
- `make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

File: charts/karpenter/templates/post-install-hook.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-post-install-hook
```
[the remark this answers is not in the document]
Comment (njtran) [reply charts/karpenter/templates/post-install-hook.yaml:4]: Think this should be fixed as of now

**Review by njtran — APPROVED** [review]

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Review by jmdeal — APPROVED** [review]
LGTM 🚀

**Pull request merged**